src/book/mod.rs

```diff
@@ -270,8 +279,16 @@ impl MDBook {
                     _ => continue,
                 };
 
-                let path = self.source_dir().join(&chapter_path);
-                info!("Testing file: {:?}", path);
+                let cp = chapter_path.to_str();
+                let cn: Option<&str> = Some(&ch.name);
+
+                if chapter.is_some() && cn != chapter && cp != chapter {
+                    info!("Skipping chapter '{}'...", ch.name);
+                    continue;
+                };
```

Comment:
Instead of using local variables like this, I still think it would be clearer to use something like the suggestion I posted before. The local variables create an indirection where you then have to refer to other parts of the code to figure out what they mean. Using the values directly I think makes it a little clearer as to what is being checked. 

Also, after testing this out a bit, I feel like the output is overwhelmed with the "skipping chapter" text. I'm thinking it might be better to reduce that to a `debug` message to reduce the noise.

Perhaps something like this:

```suggestion
                if let Some(chapter) = chapter {
                    if ch.name != chapter && chapter_path.to_str() != Some(chapter) {
                        debug!("Skipping chapter '{}'...", ch.name);
                        continue;
                    }
                }
```

Reply:
The `skipping chapter` prompts are how you find the chapter names so you can write `mdbook test -c ?` for example then you find the chapter name you want to test and run the specific command.  But I could make it so that it looks for the specific string "?" so it only prints in that case.

Reply:
I pushed the if-check looking for command line arg "?" instead and I added that to the help text to make it discoverable.

Reply:
Oh, and I see now how you are using `if let Some` to remove the need for local variables, so I pushed that fix too.